### PR TITLE
Added: Currency type

### PR DESCRIPTION
**Pull request by pitiphong-p**
This Currency will help us to work on the currency conversion between real-world amount and Omise amount more easily and consistency.

### CODE REVIEW

File: OmiseSwift/Balance.swift [hunk 1]
```diff
@@ -13,9 +13,9 @@ public class Balance: ResourceObject {
         set { set("total", Int64Converter.self, toValue: newValue) }
     }
     
-    public var currency: String? {
-        get { return get("currency", StringConverter.self) }
-        set { set("currency", StringConverter.self, toValue: newValue) }
+    public var currency: Currency? {
+        get { return get("currency", StringConverter.self).flatMap(Currency.init(code:)) }
```
Comment (chakrit) [inline OmiseSwift/Balance.swift:17]: @pitiphong-p currency is used in many places, maybe it warrants a `CurrencyConverter.self`

implementation is quite simple.

File: OmiseSwift/Currency.swift [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    case .THB:
+      return centBasedCurrencyFactor
+    case .JPY:
+      return identicalBasedCurrencyFactor
+    case .Custom(code: _, factor: let factor):
+      return factor
+    }
+  }
+  
+  public func convertFromOmiseAmount(value: Int64) -> Double {
```
Comment (chakrit) [inline OmiseSwift/Currency.swift]: @pitiphong-p `OmiseAmount` -> `subunit`

File: OmiseSwift/Balance.swift (hunk 1 above)
Comment (pitiphong-p) [reply OmiseSwift/Balance.swift:17]: We can't make `CurrencyConverter` for now. Converter requires that its `Target` type must be a subclass of `NSObject`. Currency is an associated value enum so we're out of luck for now.
I already changed the Converter's Target type to `Any` in the swift-3 branch which should allow us to do `CurrencyConverter`

File: OmiseSwift/Currency.swift (hunk 2 above)
Comment (pitiphong-p) [reply OmiseSwift/Currency.swift]: What type of `factor` should be? Int or Double

File: OmiseSwift/Balance.swift (hunk 1 above)
Comment (chakrit) [reply OmiseSwift/Balance.swift:17]: @pitiphong-p 👍

File: OmiseSwift/Currency.swift (hunk 2 above)
Comment (chakrit) [reply OmiseSwift/Currency.swift]: @pitiphong-p `Int` should be fine.

Comment (chakrit) [reply OmiseSwift/Currency.swift]: @pitiphong-p we can check in that json files robin posted if you're not sure.

Comment (pitiphong-p) [reply OmiseSwift/Currency.swift]: Ok I will go with Int

**Pull request merged**